lib/rmt/mirror.rb

```diff
@@ -100,18 +100,9 @@ def mirror_metadata
       end
     end
 
-    primary_files = []
-    deltainfo_files = []
-
-    repomd_parser = RMT::Rpm::RepomdXmlParser.new(local_filename)
-    repomd_parser.parse
-
-    metadata_files = []
-    repomd_parser.referenced_files.each do |reference|
-      metadata_files << reference
-      primary_files << reference.location if (reference.type == :primary)
-      deltainfo_files << reference.location if (reference.type == :deltainfo)
-    end
+    metadata_files = RepomdParser::RepomdXmlParser.new(local_filename).parse
+    primary_files = metadata_files.select { |reference| reference.type == :primary }
+    deltainfo_files = metadata_files.select { |reference| reference.type == :deltainfo }
```

Comment:
Just a small addition: Would it make sense to abstract the selection of `primary_files` and `deltainfo_files` into a method? If we need to select the files in multiple locations and the criteria to select them change we only need to change the selection there.

Reply:
HI, @jmammen. I thought of two ways for doing what you suggested:

1. Abstracting these lines into two different methods (e.g.: `select_primary_files` and `select_deltainfo`files` or;
1. Creating just one method which receive the file type as a parameter (e.g.: `select_metadata_files_by_type(:primary)`).

The second option is more flexible, but it would be prone to mistyping: should someone mistype the filetype symbol (e.g.: `:delta_info`), the method would fail silently (it'd return no files and it'd also not raise any exceptions). Also, if we needed to select the files in multiple locations, it would incur typing the file type symbol in as many locations as needed.

The first option would incur duplicated code, though it would leave room for improving/modifying the selections separately in the future, if needed.

I'm going with the first option, but I'd appreciate your thoughts.

Reply:
I followed your suggestions on 687711c.

Reply:
> If we need to select the files in multiple locations and the criteria to select them change we only need to change the selection there.

We will never need to, though. And when and if we do -- then we can make those changes.

> Just a small addition: Would it make sense to abstract the selection of `primary_files` and `deltainfo_files` into a method?

It is really questionable if having one-line methods is better than having one-line idiomatic Ruby code.

Reply:
HI @ikapelyukhin, you can see the extraction of the selection logic [here](https://github.com/SUSE/rmt/commit/687711c45d3cf2374480e924623b1d5b4e6843aa#diff-f9f254fce3569c11e6eac76b1a4d28aaR114-R121).

> It is really questionable if having one-line methods is better than having one-line idiomatic Ruby code.

Yes, I agree. In this case, I just wrapped the one-line idiomatic Ruby code into a method, so you can read the code block without needing to worry about how the selection is done ("low level" details). I know clean code can be too opinionated sometimes and usually it is a matter of personal taste.

Do you think the changes impaired the readability?

Reply:
> Do you think the changes impaired the readability?

I do, and I'm annoyed by the faulty logic that has led to that situation (`It hypothetically might be useful in the future if certain special conditions are met` -- that's not a valid way to make engineering decisions). But I'm not your boss, so you do you :slightly_smiling_face:

Reply:
Sure, it makes sense, it's not an optimization but it's premature. I reverted the changes in d09b486.